src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -91,13 +97,23 @@ private void init() {
     @Scheduled(fixedDelay = 30000)
     private void updateBrokers() {
         if (kafkaProperties != null) {
+            final Properties oldProperties = getKafkaProperties();
+
             final List<Broker> brokers = fetchBrokers();
             if (!brokers.isEmpty()) {
```

Comment:
Same here, haven't touched that code. :) And I agree with you, if there is no single broker that's a bigger issue not solved by just keeping the old brokers.

Maybe the idea was that if the list of brokers is empty the reason is that the reading from Zookeeper was faulty. So rather stay with the current list. But I don't know.